pandas/core/format.py

```diff
@@ -613,8 +613,12 @@ def get_col_type(dtype):
             name = any(self.frame.columns.names)
             for i, lev in enumerate(self.frame.index.levels):
                 lev2 = lev.format(name=name)
-                width = len(lev2[0])
-                lev3 = [' ' * width] * clevels + lev2
+                blank = ' ' * len(lev2[0])
+                lev3 = [blank] * clevels
+                for level_idx, group in itertools.groupby(
+                        self.frame.index.labels[i]):
+                    count = sum(1 for _ in group)
```

Comment:
Changed it back to the original form (which seems unexpectedly faster from a simple test).